File: client/mobile/Sudoo/app/src/main/java/com/sudo248/sudoo/domain/entity/order/OrderSupplierInfo.kt
package com.sudo248.sudoo.domain.entity.order

import com.sudo248.base_android.base.ItemDiff
import java.time.LocalDateTime

data class OrderSupplierInfo(
    val orderSupplierId: String,
    val supplierId: String,
    val supplierName: String,
    val supplierAvatar: String,
    val supplierBrand: String,
    val supplierContactUrl: String,
    var status: OrderStatus,
    val expectedReceiveDateTime: LocalDateTime,
    val totalPrice: Double,
    val createdAt: LocalDateTime,
    val orderCartProduct: OrderCartProduct,
    val totalProduct: Int,
) : ItemDiff {
    override fun isContentTheSame(other: ItemDiff): Boolean {
        return other is OrderSupplierInfo && other == this
    }

    override fun isItemTheSame(other: ItemDiff): Boolean {
        return other is OrderSupplierInfo &&
                other.orderSupplierId == this.orderSupplierId &&
                other.supplierId == this.supplierId &&
                other.status == this.status
    }

}

File: client/web/sudoo/lib/domain/model/order/order_supplier_info.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:sudoo/domain/model/order/order_status.dart';

part 'order_supplier_info.g.dart';

@JsonSerializable()
class OrderSupplierInfo {
  final String orderSupplierId;
  final String supplierId;
  final String supplierName;
  final String userFullName;
  final String userPhoneNumber;
  final String? paymentType;
  final DateTime? paymentDateTime;
  OrderStatus status;
  final String address;
  final DateTime expectedReceiveDateTime;
  final double totalPrice;
  final DateTime createdAt;

  OrderSupplierInfo(
    this.orderSupplierId,
    this.supplierId,
    this.supplierName,
    this.userFullName,
    this.userPhoneNumber,
    this.paymentType,
    this.paymentDateTime,
    this.status,
    this.address,
    this.expectedReceiveDateTime,
    this.totalPrice,
    this.createdAt,
  );

  factory OrderSupplierInfo.fromJson(Map<String, dynamic> json) => _$OrderSupplierInfoFromJson(json);

  Map<String, dynamic> toJson() => _$OrderSupplierInfoToJson(this);
}
